exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricExporter.java

```diff
@@ -96,62 +93,73 @@ private static MetricExporter createWithCredentials(
+          }
+          break;
+        default:
+          logger.error(
+          "Metric type {} not supported. Only gauge and cumulative types are supported.",
+          metricData.getType());
+          continue;
       }
-      MetricData.Point metricPoint = metricData.getPoints().iterator().next();
 
-      MetricDescriptor descriptor = mapMetricDescriptor(metricData, metricPoint);
-      if (descriptor == null) {
-        continue;
-      }
-      metricServiceClient.createMetricDescriptor(
-          CreateMetricDescriptorRequest.newBuilder()
-              .setName(PROJECT_NAME_PREFIX + projectId)
-              .setMetricDescriptor(descriptor)
-              .build());
-
-      MetricWithLabels updateKey =
-          new MetricWithLabels(descriptor.getType(), metricPoint.getLabels());
-
-      // Cloud Monitoring API allows, for any combination of labels and
-      // metric name, one update per WRITE_INTERVAL seconds
-      long pointCollectionTime = metricPoint.getEpochNanos();
-      if (lastUpdatedTime.containsKey(updateKey)
-          && pointCollectionTime
-              <= lastUpdatedTime.get(updateKey) / NANO_PER_SECOND + WRITE_INTERVAL_SECOND) {
-        continue;
+      for (final MetricDescriptor descriptor: builder.getDescriptors()) {
+        // TODO: limit this ONCE per JVM.
+        metricServiceClient.createMetricDescriptor(
+            CreateMetricDescriptorRequest.newBuilder()
+                .setName(PROJECT_NAME_PREFIX + projectId)
+                .setMetricDescriptor(descriptor)
+                .build());
       }
 
-      Metric metric = mapMetric(metricPoint, descriptor.getType());
-      Point point = mapPoint(metricData, metricPoint, updateKey, lastUpdatedTime);
-      if (point == null) {
-        continue;
-      }
-
-      allTimesSeries.add(
-          TimeSeries.newBuilder()
-              .setMetric(metric)
-              .addPoints(point)
-              .setResource(mapResource(projectId))
-              .setMetricKind(descriptor.getMetricKind())
-              .build());
-    }
-    createTimeSeriesBatch(metricServiceClient, ProjectName.of(projectId), allTimesSeries);
-    if (allTimesSeries.size() < metrics.size()) {
+      // TODO: Filter metrics by last updated time....
+      // MetricWithLabels updateKey =
+      //     new MetricWithLabels(descriptor.getType(), metricPoint.getLabels());
+
+      // // Cloud Monitoring API allows, for any combination of labels and
+      // // metric name, one update per WRITE_INTERVAL seconds
+      // long pointCollectionTime = metricPoint.getEpochNanos();
+      // if (lastUpdatedTime.containsKey(updateKey)
+      //     && pointCollectionTime
+      //         <= lastUpdatedTime.get(updateKey) / NANO_PER_SECOND + WRITE_INTERVAL_SECOND) {
+      //   continue;
+      // }
+    }    
+    List<TimeSeries> series = builder.getTimeSeries();
+    createTimeSeriesBatch(metricServiceClient, ProjectName.of(projectId), series);
```

Comment:
Where do GCM API errors get surfaced? It doesn't look like `createTimeSeriesBatch()` throws. API errors seem like the most reasonable case to `return CompletableResultCode.ofFailure()`

Reply:
I need to refactor all of that code.  I opened up an issue to track it, but yeah... it's not clear what happens right now.

Even more, we may want to use an asynchronous client to avoid blocking, as we return a completable result (i.e an async observable).  We'll fix this up more post-1.0 as it won't break existing usage, just make it better.